src/hpguppi_ata_blade_obs_rawdisk_thread.c

```diff
@@ -358,36 +373,66 @@ static void *run(hashpipe_thread_args_t * args)
         if(directio) {
           open_flags |= O_DIRECT;
         }
-        for(i = 0; i < nbeams; i++){
-          sprintf(fname, "%s-beam%04d.%04d.raw", pf.basefilename, i, filenum);
+        if(nbeams == 0){
+          sprintf(fname, "%s-beam.raw", pf.basefilename);
 
           fprintf(stderr, "Opening first raw file '%s' (directio=%d)\n", fname, directio);
-          fdraws[i] = open(fname, open_flags, 0644);
-          if (fdraws[i]==-1) {
+          fdraws[0] = open(fname, open_flags, 0644);
+          if (fdraws[0]==-1) {
             hashpipe_error(thread_name, "Error opening file.");
             pthread_exit(NULL);
           }
-        }
+        } else{
+          for(i = 0; i < nbeams; i++){
+            sprintf(fname, "%s-beam%04d.%04d.raw", pf.basefilename, i, filenum);
 
+            fprintf(stderr, "Opening first raw file '%s' (directio=%d)\n", fname, directio);
+            fdraws[i] = open(fname, open_flags, 0644);
+            if (fdraws[i]==-1) {
+              hashpipe_error(thread_name, "Error opening file.");
+              pthread_exit(NULL);
+            }
+          }
+        }
       }
 
       /* See if we need to open next file */
       if (block_count >= blocks_per_file) {
         filenum++;
-        for(i = 0; i < nbeams; i++){
-          close(fdraws[i]);
-          sprintf(fname, "%s-beam%04d.%04d.raw", pf.basefilename, i, filenum);
+        //single file
+        if(nbeams==0){
+          close(fdraws[0]);
+          sprintf(fname, "%s.raw", pf.basefilename);
 
           open_flags = O_CREAT|O_WRONLY;//|O_SYNC;
           if(directio) {
             open_flags |= O_DIRECT;
           }
           fprintf(stderr, "Opening next raw file '%s' (directio=%d)\n", fname, directio);
-          fdraws[i] = open(fname, open_flags, 0644);
-          if (fdraws[i]==-1) {
+          fdraws[0] = open(fname, open_flags, 0644);
+          if (fdraws[0]==-1) {
             hashpipe_error(thread_name, "Error opening file.");
             pthread_exit(NULL);
           }
+
+        }
+        //multiple files
+        else{
+          for(i = 0; i < nbeams; i++){
```

Comment:
would be neater to have an nfdraws and make sections universal between the 2 scenarios